pkg/upgrade/control_plane.go

```diff
@@ -482,12 +482,37 @@ func (u *ControlPlaneUpgrader) updateMachine(replacementKey ctrlclient.ObjectKey
 		return true, nil
 	})
 	if err != nil {
-		return errors.Wrapf(err, "timed out deleting machine %s/%s", machine.Namespace, machine.Name)
+		return errors.Wrapf(err, "timed out asking to delete machine %s/%s", machine.Namespace, machine.Name)
+	}
+
+	log.Info("Waiting for machine to be deleted")
+	err = wait.PollImmediate(deleteMachineInterval, deleteMachineTimeout, func() (bool, error) {
```

Comment:
Would it make sense to use foreground deletion on the Machine instead of polling here?

Reply:
I think we still have to poll - doesn't the `DELETE` http request return immediately no matter what you use for propagation policy? The GC controller is responsible for dealing with foreground vs background deletion of dependents.

Reply:
I want to say that the call blocks until it has been deleted, at least that is the behavior that I remember from implementing pivot for `clusterctl`

Reply:
I tried this out and the call did not block:

https://gist.github.com/chuckha/032086240945cfd3281b6137d7b4d1da

If this sample code waits for deletion to complete, there should be a panic.

Reply:
Yeah I'm 99.999% certain that you have to add code to block if you want to wait